Source/Core/Core/IOS/Network/KD/VFF/VFFUtil.cpp

```diff
@@ -200,14 +193,27 @@ static ErrorCode WriteFile(const std::string& filename, const std::vector<u8>& t
     return WC24_ERR_FILE_OPEN;
   }
 
-  // FIXME: Handle written_size < tmp_buffer.size()
-  u32 written_size;
-  const auto write_error_code =
-      f_write(&dst, tmp_buffer.data(), static_cast<UINT>(tmp_buffer.size()), &written_size);
-  if (write_error_code != FR_OK)
+  size_t size = tmp_buffer.size();
+  while (size > 0)
   {
-    ERROR_LOG_FMT(IOS_WC24, "Failed to write file {} to VFF {}", filename, write_error_code);
-    return WC24_ERR_FILE_WRITE;
+    u32 chunk_size = std::min(size, tmp_buffer.size());
+
+    u32 written_size;
+    const auto write_error_code = f_write(&dst, tmp_buffer.data(), chunk_size, &written_size);
+    if (write_error_code != FR_OK)
+    {
+      ERROR_LOG_FMT(IOS_WC24, "Failed to write file {} to VFF {}", filename, write_error_code);
+      return WC24_ERR_FILE_WRITE;
+    }
+
+    if (written_size != chunk_size)
+    {
+      ERROR_LOG_FMT(IOS_WC24, "Failed to write bytes of file {} to VFF ({} != {})", filename,
+                    written_size, chunk_size);
+      return WC24_ERR_FILE_WRITE;
+    }
+
+    size -= chunk_size;
```

Comment:
Uh, I think you missed the part where the `tmp_buffer` in FatFsUtil was just used as a temporary storage and deliberately sized to be small enough so that `min(size, u32(buffer.size()))` always resulted in a value that fits into an `u32`. The problem we're dealing with here is that the third parameter in `f_write` only takes an `u32`, so if `tmp_buffer.size()` is bigger than that we need to write in chunks. I'd suggest instead something like:

```
  size_t size = tmp_buffer.size();
  size_t offset = 0;
  while (size > 0)
  {
    constexpr size_t MAX_CHUNK_SIZE = 32768;
    u32 chunk_size = static_cast<u32>(std::min(size, MAX_CHUNK_SIZE));

    u32 written_size;
    const auto write_error_code =
        f_write(&dst, tmp_buffer.data() + offset, chunk_size, &written_size);
    if (write_error_code != FR_OK)
    {
      ERROR_LOG_FMT(IOS_WC24, "Failed to write file {} to VFF {}", filename, write_error_code);
      return WC24_ERR_FILE_WRITE;
    }

    if (written_size != chunk_size)
    {
      ERROR_LOG_FMT(IOS_WC24, "Failed to write bytes of file {} to VFF ({} != {})", filename,
                    written_size, chunk_size);
      return WC24_ERR_FILE_WRITE;
    }

    size -= chunk_size;
    offset += chunk_size;
  }
```